src/lib/Permission/LimitationResolver.php

```diff
@@ -0,0 +1,170 @@
+
+    public function __construct(
+        ContentService $contentService,
+        ContentTypeService $contentTypeService,
+        LanguageService $languageService,
+        LocationService $locationService,
+        LookupLimitationsTransformer $lookupLimitationsTransformer,
+        PermissionResolver $permissionResolver
+    ) {
+        $this->contentService = $contentService;
+        $this->contentTypeService = $contentTypeService;
+        $this->languageService = $languageService;
+        $this->locationService = $locationService;
+        $this->lookupLimitationsTransformer = $lookupLimitationsTransformer;
+        $this->permissionResolver = $permissionResolver;
+    }
+
+    public function getContentCreateLimitations(Location $parentLocation): LookupLimitationResult
+    {
+        $contentInfo = $parentLocation->getContentInfo();
+        $contentType = $contentInfo->getContentType();
+        $contentCreateStruct = $this->contentService->newContentCreateStruct($contentType, $contentInfo->getMainLanguageCode());
+        $contentCreateStruct->sectionId = $contentInfo->getSection();
+        $locationCreateStruct = $this->locationService->newLocationCreateStruct($parentLocation->id);
+
+        $versionBuilder = new VersionBuilder();
+        $versionBuilder->translateToAnyLanguageOf($this->getActiveLanguageCodes());
+        $versionBuilder->createFromAnyContentTypeOf($this->getContentTypeIds());
+
+        return $this->permissionResolver->lookupLimitations(
+            'content',
+            'create',
+            $contentCreateStruct,
+            [$versionBuilder->build(), $locationCreateStruct],
+            [Limitation::CONTENTTYPE, Limitation::LANGUAGE]
+        );
+    }
+
+    public function getContentUpdateLimitations(Location $parentLocation): LookupLimitationResult
+    {
+        $versionBuilder = new VersionBuilder();
+        $versionBuilder->translateToAnyLanguageOf($this->getActiveLanguageCodes());
+        $versionBuilder->createFromAnyContentTypeOf($this->getContentTypeIds());
+
+        return $this->permissionResolver->lookupLimitations(
+            'content',
+            'edit',
+            $parentLocation->getContentInfo(),
+            [$versionBuilder->build(), $parentLocation],
+            [Limitation::CONTENTTYPE, Limitation::LANGUAGE]
+        );
+    }
+
+    public function getLanguageLimitations(
+        string $function,
+        ValueObject $valueObject,
+        iterable $languages = [],
+        array $targets = []
+    ): array {
+        $languages = !empty($languages) ? $languages : $this->languageService->loadLanguages();
+        $lookupLimitations = $this->permissionResolver->lookupLimitations(
+            'content',
+            $function,
+            $valueObject,
+            array_merge(
+                $targets,
+                [(new VersionBuilder())->translateToAnyLanguageOf($this->getActiveLanguageCodes($languages))->build()]
```

Comment:
I would split this line a little for readability.

Reply:
Fixed: https://github.com/ibexa/admin-ui/pull/887/files#diff-255c497bbc3d254ab25bdabdab05bca82bca0dd565a170af84c8df2e661ace87R99-R100